### ISSUE TITLE
Add pair_plot_focus

### ISSUE THREAD

**Issue by OriolAbril** [arviz-devs/arviz-plots#90]
Talking with @daniel-saunders-phil he mentioned he often wanted to generate pair plots of a specific variable against several others. I think we could have a function that makes that easy (and paves the way with visuals and computation for `plot_pair` once we have the PlotMatrix manager) because as we compare multiple variables to always the same one the facetting/mapping logic is the one in `PlotCollection`.

I was thinking the API would be similar to a regular plot with dt, group, var_names... and then a `target` (or similar) name where you provide either a string (var_name in dt input) or a DataArray.

---

pseudo example

```python
from arviz_base import load_arviz_data
from arviz_base.labels import BaseLabeller, NoVarLabeller
import arviz_plots as azp

azp.style.use("arviz-clean")
plt.rcParams["font.size"] = 8  # clean theme sets font too big

idata = load_arviz_data("centered_eight")
target_name = "mu"
pc = azp.PlotCollection.wrap(
    idata.posterior.ds[["theta"]],
    cols=["school"],
    col_wrap=4,
    plot_grid_kws={"figsize": (8, 3), "sharey": True},
)
pc.map(
    azp.visuals.scatter_x,
    "samples",
    y=idata.posterior[target_name],
    marker=".",
    alpha=.4,
    edgecolor="none",
)
pc.map(
    azp.visuals.labelled_title,
    "title",
    subset_info=True,
    labeller=NoVarLabeller(),
    ignore_aes={"color"},
    size=8,
)

pc.viz["chart"].item().suptitle(f"Pair plots of theta subsets vs {target_name}");
```

![imatge](https://github.com/user-attachments/assets/650d9bc8-9bb3-44ff-a1e5-3a4d6ef85d89)

**Comment by daniel-saunders-phil** [comment]
I spent some time playing with this and it's really handy. I wanted to add two features but couldn't find the path forward. How might I color different chains differently? Also, how might I flag divergences?

**Comment by The-Broken-Keyboard** [comment]
@aloctavodia I will do this .... please assign it to me

**Comment by The-Broken-Keyboard** [comment]
@aloctavodia, is this correct approach to display divergent data ( since I was trying to figure out that how to flag divergence ) ?

The example plot which I am sharing is a pair plot of "mu" and "tau" , "mu" is on x-axis and "tau" is on y-axis....different colours denote different chains and black coloured circles denote divergent data points

![Image](https://github.com/user-attachments/assets/076c2af6-1940-4573-bdb5-8b8d4d4dc5f7)

please let me know your opinion, so that I would proceed ahead accordingly...

**Comment by aloctavodia** [comment]
Yes, you can use a different colour for divergences. In most common use cases, chains will not be distinguished by color.

**Comment by The-Broken-Keyboard** [comment]
Can I start working on  pair_plot_focus and pair_plot ? Now Since PlotMatrix manager has been added, so we can use PlotCollection's map to implement pair_plot_focus ( one variable against others ) and we can use PlotMatrix manager to implement pair_plot ( all variables against all in pairs ). Please let me know if I am thinking in write direction....
@aloctavodia @OriolAbril @rohanbabbar04 .

**Comment by OriolAbril** [comment]
You can get started with these yes. PlotMatrix will probably need some extra work still and more tests but I wanted to merge what I had so far to start unblocking things and to avoid it getting git conflicts over time.
